### ISSUE TITLE
MAINT: Replace release url with GH releases

### ISSUE THREAD

**Issue by larsoner** [openmeeg/openmeeg#495]
@papadop your comment in https://github.com/openmeeg/openmeeg/issues/331#issuecomment-1199444834 reminded me -- if you feel like doing some CI infrastructure work, we should replace https://files.inria.fr/OpenMEEG/download/ with adding assets directly to the GitHub releases, e.g. how it's done here:

https://github.com/mne-tools/mne-installers/releases/tag/v1.0.3
https://github.com/mne-tools/mne-installers/blob/59ca501957c2ff54f359cacd2375c9b87cbd58ed/.github/workflows/build.yml#L436-L500

Basically we "just" need to have a CI run that builds the binaries for each platform (we are probably close on the 18.04 run for linux, or the cibuildwheel runs for even better backward compat), outputs them as an artifact, then use the action to attach them to the release

- [ ] Replace https://files.inria.fr/OpenMEEG/download with GH releases that are build on commit/release
- [ ] Update `openmeeg.github.io`

**Comment by larsoner** [comment]
We might want to do a build separate from `cibuildwheel` that uses all static linking. The `cibuildwheel` uses dynamic and then vendors the libraries.

Alternatively, we can static linking working properly in `cibuildwheel`, but this was a pain at some point when I tried it.

Either way I think we get close to what we want with `-DENABLE_PACKAGING=ON` and `--target package` in our `cmake` command.
